Sources/Sentry/SentryOutOfMemoryScopeObserver.m

```diff
@@ -0,0 +1,115 @@
+#import "SentryOutOfMemoryScopeObserver.h"
+#import <SentryBreadcrumb.h>
+#import <SentryFileManager.h>
+#import <SentryLog.h>
+
+@interface
+SentryOutOfMemoryScopeObserver ()
+
+@property (strong, nonatomic) SentryFileManager *fileManager;
+@property (strong, nonatomic) NSFileHandle *fileHandle;
+@property (nonatomic) NSInteger maxBreadcrumbs;
+@property (nonatomic) NSInteger breadcrumbCounter;
+@property (strong, nonatomic) NSString *activeFilePath;
+
+@end
+
+@implementation SentryOutOfMemoryScopeObserver
+
+- (instancetype)initWithMaxBreadcrumbs:(NSInteger)maxBreadcrumbs
+                           fileManager:(SentryFileManager *)fileManager
+{
+    if (self = [super init]) {
+        self.maxBreadcrumbs = maxBreadcrumbs;
+        self.fileManager = fileManager;
+        self.breadcrumbCounter = 0;
+
+        [self switchFileHandle];
+    }
+
+    return self;
+}
+
+- (void)dealloc
+{
+    [self.fileHandle closeFile];
```

Comment:
`m`: We are missing a test for this line of code here.

Reply:
There is no way to see if a file handle is open or not, I don't think we can test this.

Reply:
What happens if we don't close the file?

Reply:
Nothing much.

Reply:
I'm fine then without adding a specific test for this.